lighthouse-core/gather/driver.js

```diff
@@ -506,10 +543,17 @@ class Driver {
     const maxWaitMs = (options.flags && options.flags.maxWaitForLoad) ||
         Driver.MAX_WAIT_FOR_FULLY_LOADED;
 
-    return this.sendCommand('Page.enable')
+    const redirectTracking = {url};
+
+    return this._beginNetworkMonitoring(redirectTracking)
+      .then(_ => this.sendCommand('Page.enable'))
       .then(_ => this.sendCommand('Emulation.setScriptExecutionDisabled', {value: disableJS}))
       .then(_ => this.sendCommand('Page.navigate', {url}))
-      .then(_ => waitForLoad && this._waitForFullyLoaded(pauseAfterLoadMs, maxWaitMs));
+      .then(_ => waitForLoad && this._waitForFullyLoaded(pauseAfterLoadMs, maxWaitMs))
+      .then(_ => {
+        this._endNetworkMonitoring();
+        return redirectTracking.url;
```

Comment:
went more or less with this, but only internal to driver. Didn't do a `getLatestUrl()` as driver is only semi-stateful, so it's not always clear that's what "latest" would be referring to, but got rid of all the options.url business. `gotoURL` still resolves on the final URL since during that method is the only time redirects are being tracked.